### PR TITLE
[SIDE-2236] Fix hotkey matching logic

### PR DESCRIPTION
**Pull request by richseviora**
This PR:

- Reworks the Hotkey matching logic so that it prevents matches when the actual keypress is a superset of the requested keypress.
- Adds some tests to validate the above.

### CODE REVIEW

File: ldk/go/keyboardService_test.go [hunk 1]
```diff
@@ -50,7 +51,7 @@ func TestKeyModifiers(t *testing.T) {
 				for j, fn := range funcs {
 					fnName := runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
 					if i == j {
-						t.Logf("testing %d -> %s", bit, fnName)
+						// t.Logf("testing %d -> %s", bit, fnName)
```
Comment (cshields-oliveai) [inline ldk/go/keyboardService_test.go:54]: Is the log needed?

**Review by cshields-oliveai — APPROVED** [review]
Looks Good. Reminder - here is the associated Github Issue: https://github.com/open-olive/loop-development-kit/issues/163

File: ldk/go/keyboardService_test.go [hunk 2]
```diff
@@ -64,3 +65,73 @@ func TestKeyModifiers(t *testing.T) {
 		})
 	}
 }
+
+func TestHotKeyMatch(t *testing.T) {
```
Comment (cshields-oliveai) [inline ldk/go/keyboardService_test.go:69]: Thank you for adding these.

**Review by RusGraf — APPROVED** [review]

File: ldk/go/keyboardService_test.go (hunk 1 above)
Comment (richseviora) [reply ldk/go/keyboardService_test.go:54]: Honestly I think I'd rather keep it commented out; it makes the test output really noisy.

**Pull request merged**